js/main.js

```diff
@@ -34,32 +34,73 @@ function submitInfo(event) {
     notes,
   };
   // assigns value of nextEntryId property of data object to new property entryId and adds to $formData
-  formData.entryId = data.nextEntryId;
-  // increments value of nextEntryId
-  data.nextEntryId = data.nextEntryId + 1;
-
-  // adds object to beginning of array
-  data.entries.unshift(formData);
-
-  // Resets image src
-  $userCurrentImg.src = 'images/placeholder-image-square.jpg';
-  // resets form
-  $entryForm.reset();
-
+  if (data.editing === null) {
+    // alert('if works');
+    formData.entryId = data.nextEntryId;
+    // increments value of nextEntryId
+    data.nextEntryId = data.nextEntryId + 1;
+
+    // adds object to beginning of array
+    data.entries.unshift(formData);
+
+    // Resets image src
+    $userCurrentImg.src = 'images/placeholder-image-square.jpg';
+    // resets form
+    $entryForm.reset();
+
+    // prepends list to ul
+    $list.prepend(renderEntry(formData));
+  } else {
+    // code for when editing goes here
+    const editedEntryId = data.editing.entryId;
+
+    // update data for edited entry
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === editedEntryId) {
+        data.entries[i].title = formData.title;
+        data.entries[i].url = formData.url;
+        data.entries[i].notes = formData.notes;
+        data.entries[i].entryId = editedEntryId;
+      }
+    }
+
+    // data.entries[i].entryId === pickedEntryId
+
+    // find corresponding DOM element for edited entry
+    const $editedEntry = document.querySelector(
+      `[data-entry-id="${editedEntryId}"]`
+    );
+    if ($editedEntry) {
+      // render new DOM tree
+      for (let i = 0; i < data.entries.length; i++) {
+        if (data.entries[i].entryId === editedEntryId) {
+          const $newRenderedEntry = renderEntry(data.entries[i]);
+          $list.replaceChild($newRenderedEntry, $editedEntry);
+        }
+      }
+    }
```

Comment:
You're already doing this loop & if statement a few lines above, so you should just utilize that one rather than do this again